### ISSUE TITLE
bug: inter-account transfers not excluded from income/expense totals

### ISSUE THREAD

**Issue by araujobernardo** [araujobernardo/finance-analyser#766]
## Bug Report

**Reported during**: Manual testing session (2026-05-24)

## Symptoms

Inter-account transfers (e.g. moving money from a cheque account to savings) are being counted as regular income or expenses in the Dashboard totals, spending category chart, and budget calculations. They should be excluded, as they were before the FA-CORE-001 migration.

## Root Cause

`src/hooks/useFileUpload.ts` hardcodes `isTransfer: false` on every transaction imported from a CSV file (line 110):

```ts
const payload = {
  transactions: categorised.map((t) => ({
    ...
    isTransfer: false,      // ← always false; transfer detection is missing
    isManualTransfer: false,
  })),
};
```

The `DashboardPage` correctly filters out transfers in its calculations (`!t.isTransfer` guards at lines 127, 169, 220), but since no transaction is ever flagged as a transfer, nothing is excluded.

## What the old system did

The pre-migration pfa-v3 localStorage system had transfer detection logic that identified matching debit/credit pairs across accounts (same amount, close dates) and flagged them as transfers. That logic was not ported during FA-CORE-001.

## Acceptance Criteria

- [ ] Transactions that are inter-account transfers are flagged with `isTransfer: true` when imported via CSV
- [ ] Transfers are excluded from Dashboard income/expense totals, net spend, and spending category chart
- [ ] Transfers are excluded from Budget actual-spend calculations
- [ ] The Transactions page still shows transfers (they should be visible but clearly labelled)
- [ ] Existing `isTransfer` UI filters on TransactionsPage continue to work

## Suggested Approaches

**Option A — Heuristic auto-detection at import time**: After grouping transactions by month, detect pairs where one account has a debit of amount X and another account has a credit of the same amount X within ±1 day — flag both as `isTransfer: true`. Works automatically, no user action needed.

**Option B — Manual flagging in TransactionsPage**: Add a toggle on the Transactions page so the user can mark individual transactions as transfers. Simple to implement; requires manual effort from the user.

**Option C — CSV column mapping**: If the bank CSV has a "transfer" or "type" column, map it to `isTransfer` during parsing in `src/utils/csvParser.ts`. Bank-specific; needs per-bank handling.

Option A is the most seamless and matches the pre-migration behaviour.

## Files to Look At

- `src/hooks/useFileUpload.ts` — where `isTransfer` is hardcoded
- `src/utils/csvParser.ts` — CSV parsing logic
- `src/pages/DashboardPage.tsx` — already correctly filters `!t.isTransfer`
- `src/server/routes/transactions.ts` — import endpoint; `isTransfer` is accepted as a boolean
